### ISSUE TITLE
Content prefetching recipe

### ISSUE THREAD

**Issue by jxn** [mozilla/serviceworker-cookbook#260]
It would be nice to have a functioning recipe for content prefetching ala this demo from the Chrome docs (https://googlechrome.github.io/samples/service-worker/prefetch/index.html).

Incidentally, this example seems to work in Chrome, but not Firefox (as of FF 52), but a lot of the Chrome demos seem to be slightly broken on Firefox.

**Comment by wanderview** [comment]
@jxn What doesn't work in firefox?  The only difference I see here is that the non-cached resource gives the "corrupted content" for a bad interception in firefox when offline.  This is because the service worker is rejecting the promie passed to the `respondWith()`.

**Comment by jxn** [comment]
@wanderview that is the same thing I see, but I get the corrupted content response for both of the pre_fetched resources as well as the non-pre-fetched one.  Do you see something different?  I'm testing w/ FF 51.0a2 at the moment.@wanderview that is the same thing I see, but I get the corrupted content response for both of the pre_fetched resources as well as the non-pre-fetched one.  Do you see something different?  I'm testing w/ FF 51.0a2 at the moment.

**Comment by wanderview** [comment]
Yea, I'm testing on dev-edition as well.  Works for me here.

Are you by any chance sharing your profile between different versions of firefox?

![screenshot 2016-10-27 14 18 51](https://cloud.githubusercontent.com/assets/1256526/19779745/7aff0452-9c50-11e6-852e-9c8be0f08f90.png)

**Comment by jxn** [comment]
> Are you by any chance sharing your profile between different versions of firefox?
I am.  Is that a potential for problems?  I don't actually need Release and Nightly FF to use my profile (those are only for testing), but I didn't think it would cause problems.

**Comment by wanderview** [comment]
Yes, we do not support running a profile in a newer version and then taking it back to an older version.  It mostly works, but when disk formats change it can break.  We recently changed the disk format of the Cache API in FF52, so that it likely what you are hitting.

I've been trying to get our UX team to do a warning of some kind if we detect this, but we're not there yet.

**Comment by jxn** [comment]
@wanderview ok, thanks.  I just created a new profile and was about to post that it works.  Bah!  apologies for the bad false report.  I'd still love some examples from serviceworke.rs on this, though.

**Comment by wanderview** [comment]
FYI, I think chrome has some bug here in that its treating a thrown exception like a normal network error.  I file this:

https://bugs.chromium.org/p/chromium/issues/detail?id=660377

**Comment by delapuente** [comment]
We have this [recipe about pre-catching all the resources from an external JSON](https://serviceworke.rs/json-cache.html) file but it seems not to be simple enough.
